plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java

```diff
@@ -350,6 +353,30 @@ public Object[][] predicatePushdownProvider()
         };
     }
 
+    @Test
+    public void testHighPrecisionDecimalPredicate()
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_high_precision_decimal_predicate",
+                "(col DECIMAL(34, 0))",
+                ImmutableList.of("decimal '3141592653589793238462643383279502'"))) { // 34 precision decimal value
+            String predicateValue = "decimal '31415926535897932384626433832795028841'"; // 38 precision decimal value
+            assertThat(query("SELECT * FROM " + table.getName() + " WHERE col = " + predicateValue))
+                    .returnsEmptyResult();
```

Comment:
In this case, I was not able to apply pushdown condition.

This is the Explain of the `"SELECT * FROM " + table.getName() + " WHERE col = " + predicateValue`
```
Trino version: testversion
Fragment 0 [SINGLE]
    Output layout: [col]
    Output partitioning: SINGLE []
    Output[columnNames = [col]]
    │   Layout: [col:decimal(34,0)]
    │   Estimates: {rows: 0 (0B), cpu: 0, memory: 0B, network: 0B}
    └─ Values[]
           Layout: [col:decimal(34,0)]
           Estimates: {rows: 0 (0B), cpu: 0, memory: 0B, network: 0B}
```

Reply:
I am little but missed here, why pushdown is not happening here?
I thought Trino rounds it or pass it as it is and mongo just not able to exact match value with higher pression.
But why pushdown is not happening?

Reply:
Why is the table scan is replaced with a values node ? Is it due to different precision on both side of expression ? Can we add a comment for it

Reply:
Can we add additional test case for 34 precision decimal value ?

Reply:
> Why is the table scan is replaced with a values node ?

I am not sure about this. It seems that when the value is 38 precision then it is getting converted into null. 

This query does not even reach `MongoSession#execute`
```
"SELECT * FROM " + table.getName() + " WHERE col = " + predicateValue
```